presto-main/src/main/java/io/prestosql/sql/planner/plan/LateralJoinNode.java

```diff
@@ -92,6 +95,11 @@ public LateralJoinNode(
         this.correlation = ImmutableList.copyOf(correlation);
         this.type = type;
         this.originSubquery = originSubquery;
+
+        this.outputSymbols = Stream.of(input, subquery)
+                .flatMap(planNode -> planNode.getOutputSymbols().stream())
+                .distinct()
```

Comment:
You shouldn't distinct `input` and `subquery` symbols when lateral is of `LEFT` type (`INNER` lateral seems fine though).
Let's assume `input` produces `x` and `subquery` produces `x`. However, `x` from subquery is not really the same as `x` from `input` as left row might have no matching subquery rows (therefore right side might be null). For reference see: https://github.com/prestodb/presto/issues/8459

Reply:
In case of `LEFT` lateral and duplicate `input`, `subquery` symbols we should throw an exception.

Reply:
@sopel39 I tried changing this to
```
if (type == LEFT) {
    checkArgument(Collections.disjoint(input.getOutputSymbols(), subquery.getOutputSymbols()), "Input and subquery can have no symbols in common");
    this.outputSymbols = Stream.of(input, subquery)
                    .flatMap(planNode -> planNode.getOutputSymbols().stream())
                    .collect(toImmutableList());
}
else {
    this.outputSymbols = Stream.of(input, subquery)
            .flatMap(planNode -> planNode.getOutputSymbols().stream())
            .distinct()
            .collect(toImmutableList());
}
```
This resulted in `TestLogicalPlanner.testCorrelatedSubqueries` failing.
The query `SELECT orderkey FROM orders WHERE 3 = (SELECT orderkey)` results in LateralJoinNode which has `orderkey` in common in input and subquery.

Reply:
What's the stacktrace?

Reply:
I was trying to add the check suggested above
> In case of `LEFT` lateral and duplicate `input`, `subquery` symbols we should throw an exception.

The failure results from this check throwing an exception (assuming I have implemented the check correctly by adding 
`checkArgument(Collections.disjoint(input.getOutputSymbols(),
                                                               subquery.getOutputSymbols()), "Input and subquery can have no symbols in common");`).

Reply:
Yes. If you share full stacktrace we could check why such (incorrect) lateral is create in the first place.

Reply:
ok, the full the stacktrace is 
```
at com.google.common.base.Preconditions.checkArgument(Preconditions.java:141)
	  at io.prestosql.sql.planner.plan.LateralJoinNode.<init>(LateralJoinNode.java:102)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitLateralJoin(UnaliasSymbolReferences.java:457)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitLateralJoin(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.LateralJoinNode.accept(LateralJoinNode.java:168)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:95)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:411)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.ProjectNode.accept(ProjectNode.java:92)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:95)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitFilter(UnaliasSymbolReferences.java:403)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitFilter(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.FilterNode.accept(FilterNode.java:72)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:95)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:411)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.ProjectNode.accept(ProjectNode.java:92)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:95)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:411)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.ProjectNode.accept(ProjectNode.java:92)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:95)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:411)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.ProjectNode.accept(ProjectNode.java:92)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:95)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:411)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitProject(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.ProjectNode.accept(ProjectNode.java:92)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:95)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitOutput(UnaliasSymbolReferences.java:418)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences$Rewriter.visitOutput(UnaliasSymbolReferences.java:122)
	  at io.prestosql.sql.planner.plan.OutputNode.accept(OutputNode.java:82)
	  at io.prestosql.sql.planner.plan.SimplePlanRewriter.rewriteWith(SimplePlanRewriter.java:27)
	  at io.prestosql.sql.planner.optimizations.UnaliasSymbolReferences.optimize(UnaliasSymbolReferences.java:119)
	  at io.prestosql.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:170)
	  at io.prestosql.testing.LocalQueryRunner.createPlan(LocalQueryRunner.java:854)
```

Reply:
I think this might be a bug in `UnaliasSymbolReferences` that it unaliases symbols using correlated symbols. This shouldn't be allowed I think. Could you try to fix that? Otherwise I will take a look later.